superset/views/database/views.py

```diff
@@ -241,3 +241,152 @@ def form_post(self, form):
         flash(message, "info")
         stats_logger.incr("successful_csv_upload")
         return redirect("/tablemodelview/list/")
+
+
+class ExcelToDatabaseView(SimpleFormView):
+    form = ExcelToDatabaseForm
+    form_template = "superset/form_view/excel_to_database_view/edit.html"
+    form_title = _("Excel to Database configuration")
+    add_columns = ["database", "schema", "table_name"]
+
+    def form_get(self, form):
+        form.header.data = 0
+        form.mangle_dupe_cols.data = True
+        form.skipinitialspace.data = False
+        form.skip_blank_lines.data = True
+        form.infer_datetime_format.data = True
+        form.decimal.data = "."
+        form.if_exists.data = "fail"
+
+    def form_post(self, form):
+        database = form.con.data
+        excel_table = Table(table=form.name.data, schema=form.schema.data)
+
+        if not schema_allows_csv_upload(database, excel_table.schema):
+            message = _(
+                'Database "%(database_name)s" schema "%(schema_name)s" '
+                "is not allowed for csv uploads. Please contact your Superset Admin.",
```

Comment:
We should not refer to CSV in the Excel upload. Perhaps this could be rephrased simply as "..not allowed for uploads".